rundk/main.go

```diff
@@ -0,0 +1,192 @@
+	if err != nil {
+		log.Fatal("Error setting up the temporary directory: ", err)
+	}
+
+	// Setup docker-in-docker
+	if dindEnabled {
+		cmd.AddMount("bind", "/var/run/docker.sock", "/var/run/docker.sock")
+		// On a MAC machine, set to host networking to make DinD work. Not
+		// required on Linux.
+		if runtime.GOOS == "darwin" {
+			cmd.AddArgs("--network", "host")
+		}
+	}
+
+	// Setup gcloud credentials
+	if useLocalGcloudCredentials {
+		gcloudKey := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+		// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+		// to the container
+		if gcloudKey != "" {
+			envs["GOOGLE_APPLICATION_CREDENTIALS"] = gcloudKey
+			if _, err := os.Stat(gcloudKey); !os.IsNotExist(err) {
+				cleanup := cmd.CopyAndAddMount("bind", tmpDir, gcloudKey, gcloudKey)
+				builtUpDefers = append(builtUpDefers, cleanup)
+			}
+		}
+		// Copy and mount for gcloud default credentials to be available
+		defaultGcloudConfigPath := path.Join(os.Getenv("HOME"), ".config/gcloud")
+		if _, err := os.Stat(defaultGcloudConfigPath); !os.IsNotExist(err) {
+			cleanup := cmd.CopyAndAddMount("bind", tmpDir, defaultGcloudConfigPath, "/root/.config/gcloud")
+			builtUpDefers = append(builtUpDefers, cleanup)
+		}
+	}
+
+	// Setup kubeconfig
+	if useLocalKubeconfig {
+		kubeconfig := os.Getenv("KUBECONFIG")
+		// If KUBECONFIG is not empty, also mount the kubeconfig files to the
+		// container
+		if kubeconfig != "" {
+			envs["KUBECONFIG"] = kubeconfig
+			for _, f := range strings.Split(kubeconfig, string(os.PathListSeparator)) {
+				if _, err := os.Stat(f); !os.IsNotExist(err) {
+					cleanup := cmd.CopyAndAddMount("bind", tmpDir, f, f)
+					builtUpDefers = append(builtUpDefers, cleanup)
+				}
+			}
+		}
+		// Copy and mount for default kube context to be available
+		defaultKubeconfigPath := path.Join(os.Getenv("HOME"), ".kube")
+		if _, err := os.Stat(defaultKubeconfigPath); !os.IsNotExist(err) {
+			cleanup := cmd.CopyAndAddMount("bind", tmpDir, path.Join(os.Getenv("HOME"), ".kube"), "/root/.kube")
+			builtUpDefers = append(builtUpDefers, cleanup)
+		}
+	}
+
+	// Setup args to promote env vars from local to the Docker container
+	if err = envs.PromoteFromEnv(mandatoryEnvVars...); err != nil {
+		log.Fatal("Missing mandatory argument: ", err)
+	}
+	envs.PromoteFromEnv(optionalEnvVars...) // Optional, so don't check error
+
+	// Setup args to mount the required files and directories on the host
+	// machine to the container
+	for _, m := range mounts {
+		sourceAndTarget := strings.Split(m, ":")
+		if len(sourceAndTarget) != 2 {
+			log.Fatalf("The mount string %q must be in the format of [source:target]", m)
+		}
+		cmd.AddMount("bind", sourceAndTarget[0], sourceAndTarget[1])
+	}
+
+	// Copy and mount source code root dir
+	repoRoot, err := helpers.GetRootDir()
+	if err != nil {
+		log.Fatal("Error getting the repo's root directory: ", err)
+	}
+	cleanup := cmd.CopyAndAddMount("bind", tmpDir, repoRoot, repoRoot)
```

Comment:
you don't want to mount the repoRoot to `/go/src/...`?

Reply:
With Go mod, it does not really matter where we put the repo, it'll work under any folder.

Reply:
And this tool will need to work with other test images, which do not necessarily have their `GOPATH` set to `/go`.